core/src/main/scala/kafka/raft/RaftManager.scala

```diff
@@ -181,12 +180,7 @@ class KafkaRaftManager[T](
     val expirationTimer = new SystemTimer("raft-expiration-executor")
     val expirationService = new TimingWheelExpirationService(expirationTimer)
     val quorumStateStore = new FileBasedStateStore(new File(dataDir, "quorum-state"))
-
-    val nodeId = if (config.processRoles.contains(ControllerRole)) {
-      OptionalInt.of(config.nodeId)
-    } else {
-      OptionalInt.empty()
-    }
+    val nodeId = OptionalInt.of(config.nodeId)
```

Comment:
We did this to address this issue: https://issues.apache.org/jira/browse/KAFKA-13168. Do you think this is not needed anymore because we have this check in `KafkaConfig`?
```scala
        // nodeId must not appear in controller.quorum.voters
        require(!voterAddressSpecsByNodeId.containsKey(nodeId),
          s"If ${KafkaConfig.ProcessRolesProp} contains just the 'broker' role, the node id $nodeId must not be included in the set of voters ${KafkaConfig.QuorumVotersProp}=${voterAddressSpecsByNode  Id.asScala.keySet.toSet}")
```

Reply:
Yes, that is what I was thinking. Because we require the voter set to be consistently defined on all nodes, it seems like it is good enough to ensure that any node which starts with just the "broker" role is not among the voters. Do you agree with that?

Reply:
Okay. It is best we can do at the moment. We can revisit this when we have replicaUuid and storageId from KIP-853.